addons/flannel/0.20.2/install.sh

```diff
@@ -80,3 +92,87 @@ function flannel_weave_conflict() {
 function flannel_antrea_conflict() {
     ls /etc/cni/net.d/*antrea* >/dev/null 2>&1
 }
+
+function flannel_is_single_node() {
+    local nodecount=
+    nodecount=$(kubectl get nodes --no-headers | wc -l)
+    if [ "$nodecount" -eq 1 ]; then
+        return 0
+    else
+        return 1
+    fi
+}
+
+function weave_to_flannel() {
+    local dst="$DIR/kustomize/flannel"
+
+    logStep "Removing Weave to install Flannel"
+    remove_weave
+
+    logStep "Updating kubeadm to use Flannel"
+    flannel_kubeadm
+
+    logStep "Applying Flannel"
+    kubectl -n kube-flannel apply -k "$dst/"
+    echo "waiting for kube-flannel-ds to become healthy in kube-flannel"
+    spinner_until 240 daemonset_fully_updated "kube-flannel" "kube-flannel-ds"
+
+    logStep "Restarting kubelet"
+    systemctl stop kubelet
+    iptables -t nat -F && iptables -t mangle -F && iptables -F && iptables -X
+    echo "waiting for containerd to restart"
+    restart_systemd_and_wait containerd
+    echo "waiting for kubelet to restart"
+    restart_systemd_and_wait kubelet
+
+    logStep "Restarting pods in kube-system"
+    kubectl -n kube-system delete pods --all
+    kubectl -n kube-flannel delete pods --all
+    flannel_ready_spinner
+
+    logStep "Restarting CSI pods"
+    kubectl -n longhorn-system delete pods --all || true
+    kubectl -n rook-ceph delete pods --all || true
+    kubectl -n openebs delete pods --all || true
+
+    sleep 60
+    logStep "Restarting all other pods"
+    local ns=
+    for ns in $(kubectl get ns -o name | grep -Ev '(kube-system|longhorn-system|rook-ceph|openebs|kube-flannel)' | cut -f2 -d'/'); do
+        kubectl delete pods -n "$ns" --all
+    done
+
+    sleep 60
+    logSuccess "Migrated from Weave to Flannel"
```

Comment:
Why do we need wait 60 seconds to print `logSuccess "Migrated from Weave to Flannel"`?
Could we check the state and if all is OK then we output the `logSuccess "Migrated from Weave to Flannel"` instead?

Reply:
And how do you propose to check if everything is OK?

Reply:
By this point we already know that *flannel* is working

We're just restarting all of the pods so that they use flannel (first kube-system and flannel, then CSIs, then everything else)